src/DataSets/log.js

```diff
@@ -1,55 +1,87 @@
 import React from 'react';
 import { getInstance as getD2 } from 'd2/lib/d2';
 
+const maxLogsPerPage = 200;  // TODO: maybe make it readable from the dataStore
+const maxLogPages = 100;
 
-async function log(actionName, status, dataset) {
+
+async function log(actionName, status, datasets) {
     // Log the name of the action that has been executed, its status
     // ("success", "failed"), by whom and on which datasets.
-    const maxLogs = 1e3;
-
-    const d2 = await getD2();
-    const store = await d2.dataStore.get('dataset-configuration');
-
-    const logIndex = await store.get('logIndex').catch(() => 0);
-    const logs = await store.get('logs').catch(() => []);
-
-    const datasets = Array.isArray(dataset) ? dataset : [dataset];
+    const logs = await getLogs([0]);
+    logs.push(await makeEntry(actionName, status, datasets));
+    return setLogs(logs);
+}
 
-    const newLog = {
+async function makeEntry(actionName, status, datasets) {
+    // Return an object (a "dictionary") with the information we want to log.
+    datasets = Array.isArray(datasets) ? datasets : [datasets];
+    const user = await getD2().then(d2 => d2.currentUser);
+    return {
         date: new Date().toISOString(),
         action: actionName,
         status: status,
-        user: {displayName: d2.currentUser.name,
-               username: d2.currentUser.username,
-               id: d2.currentUser.id},
-        datasets: datasets.map(ds => ({displayName: ds.name, id: ds.id})),
+        user: {
+            displayName: user.name,
+            username: user.username,
+            id: user.id,
+        },
+        datasets: datasets.map(ds => ({
+            displayName: ds.name,
+            id: ds.id,
+        })),
     };
+}
+
+async function getLogs(pages = [0, 1]) {
+    // Return the concatenated logs for the given pages (relative to
+    // logsPageCurrent) if they exist, or an empty list otherwise.
+    // It returns null when there are no more log pages.
+    if (pages.every(n => n >= maxLogPages))
+        return null;
 
-    if (logs.length < maxLogs)
-        logs.push(newLog)
+    const store = await getStore();
+    const logsPageCurrent = await store.get('logs-page-current').catch(() => 0);
+    const pagesNames = pages.map(n => 'logs-page-' + mod(logsPageCurrent - n, maxLogPages));
+    const logs = await Promise.all(pagesNames.map(name => store.get(name).catch(() => null)));
```

Comment:
I've seen there is a way to avoid 404 on non-existing pages. The store has a method to get the current keys. Thus we could also remove the `catch`, which might mask real problems (network issues/implementation bugs). Just an idea:

```
const storeKeys = new Set(await store.getKeys());
const existingPageNames = pagesNames.filter(name => storeKeys.has(name));
const logs = await Promise.all(existingPageNames.map(name => store.get(name)));
```